windows-gaming-driver/src/driver/clientpipe/mod.rs

```diff
@@ -73,33 +74,17 @@ impl Clientpipe {
                     info!("client says that it's suspending");
                     controller.borrow_mut().ga_suspending();
                 }
-                GaCmdIn::Pong(_) => {
-                    trace!("ga pong'ed");
-                    controller.borrow_mut().ga_pong();
-                }
-                GaCmdIn::HotKey(id) => {
-                    debug!("hotkey pressed: {}", id);
-                    controller.borrow_mut().ga_hotkey(id);
-                }
-                GaCmdIn::HotKeyBindingFailed(s) => {
-                    warn!("HotKeyBinding failed: {}", s);
-                }
+                GaCmdIn::Pong(_) => controller.borrow_mut().ga_pong(),
+                GaCmdIn::HotKey(id) => controller.borrow_mut().ga_hotkey(id),
+                GaCmdIn::HotKeyBindingFailed(s) => warn!("HotKeyBinding failed: {}", s),
                 GaCmdIn::Clipboard(c) => match c.message {
-                    Some(ClipboardMessage::GrabClipboard(_)) => {
-                        debug!("windows is grabbing clipboard");
-                        controller.borrow_mut().grab_x11_clipboard();
-                    }
-                    Some(ClipboardMessage::RequestClipboardContents(data)) => {
-                        debug!("windows requested clipboard contents");
-                        controller.borrow_mut().read_x11_clipboard();
-                    }
-                    Some(ClipboardMessage::ContentsType(t)) => {
-                        unimplemented!();
-                    }
-                    Some(ClipboardMessage::ClipboardContents(buf)) => {
-                        debug!("windows responded with clipboard contents");
-                        controller.borrow_mut().respond_x11_clipboard(buf);
-                    }
+                    Some(ClipboardMessage::GrabClipboard(_)) => controller.borrow_mut().grab_x11_clipboard(),
+                    Some(ClipboardMessage::RequestClipboardContents(kind)) => match ClipboardType::from_i32(kind) {
```

Comment:
Why do we need `ClipboardType::from_i32` here? Shouldn't that already be serialized by prost?